src/auth.zig

```diff
@@ -107,6 +97,296 @@ test MTLS {
     // TODO there's likely a few more error states we should validate;
 }
 
+/// Default CookieAuth Helper uses Sha256 as the HMAC primitive.
+pub const CookieAuth = cookieAuth(Hmac.sha2.HmacSha256);
+
+pub const cookie_auth = struct {
+    /// Why are you using sha1?
+    pub const sha1 = cookieAuth(Hmac.HmacSha1);
+
+    pub const sha2 = struct {
+        pub const @"224" = cookieAuth(Hmac.sha2.HmacSha224);
+        pub const @"256" = cookieAuth(Hmac.sha2.HmacSha256);
+        pub const @"384" = cookieAuth(Hmac.sha2.HmacSha384);
+        pub const @"512" = cookieAuth(Hmac.sha2.HmacSha512);
+    };
+};
+
+/// TODO document
+pub fn cookieAuth(HMAC: type) type {
+    return struct {
+        base: ?Provider,
+        // TODO key safety
+        server_secret_key: []const u8,
+        /// Max age in seconds a session cookie is valid for.
+        max_age: usize,
+        cookie_name: []const u8 = "verse_session_secret",
+
+        /// this session buffer API is unstable and may be replaced
+        session_buffer: [ibuf_size]u8 = [_]u8{0} ** ibuf_size,
+        alloc: ?Allocator,
+        const ibuf_size = b64_enc.calcSize(HMAC.mac_length * 8);
+
+        pub const Self = @This();
+
+        pub fn init(opts: struct {
+            server_secret_key: []const u8,
+            alloc: ?Allocator = null,
+            base: ?Provider = null,
+            max_age: usize = 86400 * 365,
+        }) Self {
+            return .{
+                .server_secret_key = opts.server_secret_key,
+                .alloc = opts.alloc,
+                .base = opts.base,
+                .max_age = opts.max_age,
+            };
+        }
+
+        pub fn validateToken(hm: *HMAC, token: []const u8, user_buffer: []u8) Error![]u8 {
```

Comment:
Does this function need to be public?

Reply:
no, but I wanted people to be able to use it in their custom handler if they needed to write one